cache/cache.go

```diff
@@ -13,3 +13,135 @@
 // limitations under the License.
 
 package cache
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+var (
+	ErrDisabled  = errors.New("cache disabled")
+	ErrCompacted = errors.New("revision compacted")
+)
+
+type Cache struct {
+	cfg Config
+
+	cli *clientv3.Client
+	es  *eventSource
+	dx  *demux
+
+	stop context.CancelFunc
+	wg   sync.WaitGroup
+
+	gapSem chan struct{}
+	closed atomic.Bool
+}
+
+func New(cli *clientv3.Client, opts ...Option) (*Cache, error) {
+	cfg := defaultConfig()
+	for _, o := range opts {
+		o(&cfg)
+	}
+	if cfg.Disable {
+		return &Cache{cfg: cfg}, nil
+	}
+
+	// discover current revision
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	resp, err := cli.Get(ctx, "\x00", clientv3.WithLimit(1))
```

Comment:
Not sure why you want to get revision first. If you send to etcd Watch with rev=0 it will start from the latest revision.

Reply:
Just use latest RV from in ringbuffer